lib/stack_master/stack.rb

```diff
@@ -59,7 +59,9 @@ def self.generate(stack_definition, config)
       parameter_hash = ParameterLoader.load(parameter_files: stack_definition.all_parameter_files, parameters: stack_definition.parameters)
       template_parameters = ParameterResolver.resolve(config, stack_definition, parameter_hash[:template_parameters])
       compile_time_parameters = ParameterResolver.resolve(config, stack_definition, parameter_hash[:compile_time_parameters])
-      template_body = TemplateCompiler.compile(config, stack_definition.compiler, stack_definition.template_dir, stack_definition.template, compile_time_parameters, stack_definition.compiler_options)
+      json_patches = TemplatePatcher.load_json_patches(stack_definition.json_patch_file_paths)
+      unpatched_template_body = TemplateCompiler.compile(config, stack_definition.compiler, stack_definition.template_dir, stack_definition.template, compile_time_parameters, stack_definition.compiler_options)
```

Comment:
That would require `m*n` compilers, right? As you can apply this to SparkleFormation templates too?

Reply:
Could we just not support that? Only support YAML/JSON and simplify our implementation?

Reply:
We could, but would that introduce new edge cases for determining the compiler? My approach had been to ensure this would be its own layer for increased predictability (it always applies if there are patches, never if there are no patches).

Reply:
I think my mental model is just different. I see this as a compile, something you'd use on it's own and never in combination with other compilers.
We can determine the compiler based on if patches are available and if it's a YAML or JSON template?